### PR TITLE
[ldapjs] add missing close method to Server

### PR DESCRIPTION
**Pull request by ashpieboop**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [ldapjs line](https://github.com/ldapjs/node-ldapjs/blob/4f5c6ddd95ed959862497328c9d93555282f2bec/lib/server.js#L693) and [NodeJS net 'close' event](https://nodejs.org/api/net.html#net_event_close)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## ldapjs ([<kbd>unpkg</kbd>](https://unpkg.com/browse/ldapjs@latest/))
was missing the following properties:
1. Client
1. Server
1. dn
1. DN
1. RDN

<details>
<summary>as well as these 5 other properties...</summary>
<p>persistentSearch, PersistentSearchCache, filters, url, parseURL</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d9da6e9675c3b9175ef3fa2db55056fa4920ed49
</p>

**Review by mscottnelson — COMMENTED** [review]
Thanks for putting in this PR. This method is inherited, but ldapjs uses an old-fashioned inheritance model that is somewhat non-obvious, and on top of that uses lots of mutations making it very difficult to type. I've run into irritation around lack of types on this lib before, so your PR prompted me to look into this a touch. 

The close function here is inherited from net.Server (or tls.Server, depending), and thus has the signature:
`close(callback?: (err?: Error) => void): this;`

Rather than duplicate method types, I'm going to try and expose the underlying node types and extend them. If you wish, you could change your PR to target the specific method from net.Server including the optional Error input, but I haven't verified yet if any additional types exist on the return type (which will be some variation on either tls.Server or net.Server, which tls.Server extends).

I'll try and get a PR inspired by yours pushed shortly.

File: types/ldapjs/index.d.ts [hunk 1]
```diff
@@ -309,6 +309,8 @@ export interface Server extends EventEmitter {
      */
     listenFD(fileDescriptor: any): void;
 
+    close(callback: () => void): this;
```
Comment (mscottnelson) [inline types/ldapjs/index.d.ts:312]: My mistake. This method is wrapped, and cannot accept a callback at all. If you remove the callback, I can approve this for merge.

File: types/ldapjs/ldapjs-tests.ts [hunk 2]
```diff
@@ -151,3 +151,14 @@ let approximateFilter = new ldap.ApproximateFilter({
...
+
+
+let server = ldap.createServer();
+server.listen(1389, '127.0.0.1', () => {
+    // Do stuff
+
+    // Close the server gracefully
+    server.close(() => {
+        // Server closed
+    });
```
Comment (mscottnelson) [inline types/ldapjs/ldapjs-tests.ts:163]: Server.close(cb) will not do anything with the callback function.

File: types/ldapjs/index.d.ts (hunk 1 above)
Comment (ArisuOngaku) [reply types/ldapjs/index.d.ts:312]: But what about https://github.com/ldapjs/node-ldapjs/blob/master/lib/server.js#L693 ? It looks like it's full passing-through the method to `(net|tls).Server`. Furthermore, the callback actually gets called in the project where I use it.

Comment (mscottnelson) [reply types/ldapjs/index.d.ts:312]: I am so sorry. You are correct. I was inadvertently looking at a different version of ldapjs where this callback function is not present.

**Review by mscottnelson — APPROVED** [review]

**Comment by ArisuOngaku** [comment]
Ready to merge

File: types/ldapjs/index.d.ts (hunk 1 above)
Comment (ArisuOngaku) [reply types/ldapjs/index.d.ts:312]: np thank you for everything! :)

**Pull request merged**